tests/testthat/test_plot_nhdplus.R

```diff
@@ -311,3 +314,55 @@ test_that("comids", {
   expect_true(is(d$basin, "sf"))
 })
 
+test_that("waterbodies", {
+  testthat::skip_on_cran()
+  site <- "USGS-05428500"
+  tempd <- tempdir(check = TRUE)
+  g_temp <- file.path(tempd, "foo.gpkg")
+  d <-  nhdplusTools:::plot_nhdplus(site)
+
+  expect_equal(names(d), c("plot_bbox", "outlets", "flowline",
+                           "basin", "catchment","network_wbd",
+                           "off_network_wbd"))
+
+  d <-  nhdplusTools:::plot_nhdplus(site,flowline_only = FALSE)
+  expect_true(is(d$network_wbd, "sf"))
+  expect_true(is(d$off_network_wbd, "sf"))
+
+  bbox <- sf::st_bbox(c(xmin = -89.56684, ymin = 42.99816,
+                        xmax = -89.24681, ymax = 43.17192),
+                      crs = "+proj=longlat +datum=WGS84 +no_defs")
+
+  # With downloaded data
+  d <- nhdplusTools:::get_plot_data(bbox = bbox, flowline_only = FALSE)
+
+  expect_equal(nrow(d$off_network_wbd), 43)
+  expect_equal(nrow(d$network_wbd), 10)
+  # With Local Data (note this sample is already subset to a watershed basis)
+  # d <- nhdplusTools:::get_plot_data(bbox = bbox, streamorder = 2,
+  #                                   nhdplus_data = sample_data)
+  #
+  # expect_equal(nrow(d$off_network_wbd), 53)
+  # expect_equal(nrow(d$network_wbd), 0)
```

Comment:
This should work - but I'm not quite following why it doesn't.  I commented out for now but I think will need to figure out why this fails in testing

Reply:
If you'd like, we could just merge and I can take it from here if you think you want help.